modeshape-jcr/src/main/resources/org/modeshape/jcr/modeshape_builtins.cnd

```diff
@@ -140,6 +140,22 @@
     < 'STRING', 'URI', 'BINARY', 'LONG', 'DOUBLE', 'DECIMAL', 'BOOLEAN', 
       'DATE', 'NAME', 'PATH', 'REFERENCE', 'WEAKREFERENCE', 'UNDEFINED'
 
+// -----------------------
+// Large unordered collections
+// -----------------------
+// Abstract base type for all unordered collections.
+
+[mode:unorderedCollection] abstract
+  + * (nt:base) = nt:unstructured abort 
```

Comment:
I don't think mixins are a good idea because they are dynamic - they can be added/removed at runtime. Supporting this - changing how children are stored for nodes - would be highly complex.

I think the original idea of having 4 base types: `mode:unorderedXXXCollection` is a lot better since that means node types will stay immutable. Users are expected to extend any of the 4 types based on their scenario.